third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -3754,3 +3807,18 @@ pub(crate) fn extract_schema_access<'a>(exp: &'a EA::Exp, res: &mut Vec<&'a EA::
         _ => {},
     }
 }
+
+// returns the abilities that a field should have but not
+fn field_missing_abilities(struct_abilities: AbilitySet, field_abilities: AbilitySet) -> AbilitySet {
+    let mut missing_abilities = AbilitySet::EMPTY;
+    if struct_abilities.has_copy() && !field_abilities.has_copy() {
+        missing_abilities = missing_abilities.add(Ability::Copy);
+    }
+    if struct_abilities.has_copy() && !field_abilities.has_copy() {
+        missing_abilities = missing_abilities.add(Ability::Copy);
```

Comment:
If Vineeth is right, you need `Ability::Drop` here.

Reply:
done